buzzmoon-frontend/buzzmoon-vite/src/Components/BackendActor/backend-actor.js

```diff
@@ -164,4 +138,43 @@ export default class BackendActor {
 
   }
 
+  // checks whether a player can play a game, or can only view results
+  static async checkEntryMode(gameID) {
+    const response = await Parse.Cloud.run("checkEntryMode", {gameID});
+    return response;
+  }
+
+  // registers the entry of a player into a game on the backend
+  // creates a GameResult object in database to hold the results of this game
+  static async registerGameEntry(gameID) {
+    const response = await Parse.Cloud.run("registerGameEntry", {gameID});
+    return response;
+  }
+
+  // gets all GameResult objects in the database associated with a particular game 
+  static async fetchGameResults(gameID) {
+    const response = await Parse.Cloud.run("fetchAllGameResults", {gameID});
+    return response;
+  }
+
+  // takes an array of GameResults and constructs a table sorted by points scored
+  static resultsToStandardTable(results){
+    let rows = results.map(this.resultToStandardTableRow);
+
+    rows.sort((a,b) => (b.points - a.points));
+
+    return rows;
+  }
+
+  // takes the result data returned by a call to fetchGameResults and turns it into the data required for a table row
+  // in the standard leaderboard 
+  static resultToStandardTableRow(result) {
+    const points = result.answers.map((e) => (e.points)).reduce((acc, x) => (acc+x), 0);
```

Comment:
you could explore having a single map call (i.e making it multiline) since you're iterating over the same object, just different properties. the single map call can still return points, numCorrect, averageCelerity, name